File: src/main/java/me/sungbin/lec03/Lec03Main.java
package me.sungbin.lec03;

public class Lec03Main {
    static void main() {
        String str = """
                ABC
                TEXT
               
                """;
        System.out.println(str);

        String charString = "ABC";
        System.out.println(charString.charAt(0));
    }

    public static void printAgeIfPersonV1(Object obj) {
        if (obj instanceof Person) {
            Person person = (Person) obj;
            System.out.println(person.getAge());
        }
    }

    public static void printAgeIfPersonV2(Object obj) {
        if (obj instanceof Person person) {
            System.out.println(person.getAge());
        }
    }
}

File: src/main/kotlin/me/sungbin/lec03/Lec03Main.kt
package me.sungbin.lec03

fun main() {
    var number1: Int? = 3
    var number2: Long = number1?.toLong() ?: 0L
    var str: String = number1.toString()

    val person = Person("양성빈", 100)
    val name = person.name
    val age = person.age
    println("내 이름은 ${name}이고, 나이는 $age 입니다.")

    val log = """
        fsdfdsfsd
        fdsfdsfsdfds
        fsdfdsf
    """.trimIndent()
    println(log)

    val abc = "ABC"
    println(abc[0])
}

fun printAgeIfPersonV1(obj: Any) {
    if (obj is Person) {
        var person = obj as Person
        println(person.age)
    }
}

fun printAgeIfPersonV2(obj: Any) {
    if (obj is Person) {
        println(obj.age)
    }
}

fun printAgeIfPersonV3(obj: Any?) {
    if (obj is Person) {
        var person = obj as? Person
        println(person?.age ?: 0)
    }
}

fun printAgeIfPersonV4(obj: Any?) {
    if (obj !is Person) {
        throw IllegalArgumentException("obj must be Person")
    }
}

fun fail(message: String): Nothing {
    throw IllegalArgumentException(message)
}